server/src/main/java/org/elasticsearch/bootstrap/BootstrapChecks.java

```diff
@@ -713,4 +715,13 @@ boolean isAllPermissionGranted() {
 
     }
 
+    static class DiscoveryConfiguredCheck implements BootstrapCheck {
+        @Override
+        public BootstrapCheckResult check(BootstrapContext context) {
+            if (ClusterBootstrapService.discoveryIsConfigured(context.settings)) {
```

Comment:
This should only apply if `discovery.type` is `zen2` - particularly, `single-node` discovery shouldn't yield this warning.

Reply:
As it stands, we do not have a `discovery.type` of `zen2` :) I will wait for that before proceeding.